### ISSUE TITLE
error: invalid argument '-std=c++14' not allowed with 'C'

### ISSUE THREAD

**Issue by ryandesign** [Edrusb/DAR#29]
**Describe the bug**

dar 2.7.1.RC1 fails to build for me on macOS 10.13.6 with Apple LLVM version 9.1.0 (clang-902.0.39.2) from Xcode 9.4.1. This compiler supports C++14 but defaults to C++98 mode so configure output contains:

```
checking for c++14 support... no
checking for c++ support with -std=c++14 option set... yes
```

(I guess that second message should be changed to read "checking for c++**14** support with -std=c++14 option set".)

The build fails with:

```
depbase=`echo bnonzero.o | sed 's|[^/]*$|.deps/&|;s|\.o$||'`;\
	ccache /usr/bin/clang -DHAVE_CONFIG_H -I. -I../..  -DLIBDAR_MODE=64 -I'./../libdar' -DDAR_LOCALEDIR=\"/opt/local/share/locale\" -DDAR_SYS_DIR=\"\"  -I/opt/local/include   -std=c++14   -pipe -Os -Werror=implicit-function-declaration -arch x86_64 -MT bnonzero.o -MD -MP -MF $depbase.Tpo -c -o bnonzero.o bnonzero.c &&\
	mv -f $depbase.Tpo $depbase.Po
error: invalid argument '-std=c++14' not allowed with 'C'
make[3]: *** [bnonzero.o] Error 1
```

If however I configure with `-std=c++14` in CXXFLAGS then configure output contains:

```
checking for c++14 support... yes
```

and the build succeeds. This suggests that when the configure script determines that it must add `-std=c++14`, it is adding it to more than just CXXFLAGS and maybe it shouldn't do that.

**To Reproduce**
compile dar 2.7.1.RC1 with a version of clang that supports C++14 (e.g. open source clang 3.5 and later) but does not default to it (e.g. open source clang less than 6.0) (open source clang 3.4 supports C++14 but only with the `-std=c++1y` flag not the `-std=c++14` flag)

**Expected behavior**
Successful build

**Screenshots**
n/a

**Software used:**
dar 2.7.1.RC1 source code from https://dar.edrusb.org/dar.linux.free.fr/Interim_releases/
compiled with Apple LLVM version 9.1.0 (clang-902.0.39.2) from Xcode 9.4.1

**Operating system used:**
macOS 10.13.6

**Additional context**
n/a

**Comment by Edrusb** [comment]
Yep, I've fixed that too in 2.7.1.RC2 [here](https://dar.edrusb.org/dar.linux.free.fr/Interim_releases/) 
Thanks for your feedback!

**Comment by Edrusb** [comment]
the just release 2.7.1 should have this fixed now, closing this issue

**Comment by ryandesign** [comment]
> when the configure script determines that it must add `-std=c++14`, it is adding it to more than just CXXFLAGS and maybe it shouldn't do that

This problem still exists in 2.7.1.

**Comment by Edrusb** [comment]
yep I missed fixing it, reopening the issue.

**Comment by Edrusb** [comment]
I could reproduce the problem, 2.7.2.RC1 (release candidate) has a fix for that, 2.7.2 will be available in a few days

**Comment by Edrusb** [comment]
2.7.2 should be OK with that problem now, if it was still not the case :
- first you would see me really sorry! 
- second thanks to reopen this issue
Of course the issue is fixed for you too, you are welcome to let me know :)
